civis/base.py

```diff
@@ -93,8 +94,9 @@ def _make_request(self, method, path=None, params=None, data=None,
         url = self._build_path(path)
 
         with self._lock:
-            response = self._session.request(method, url, json=data,
-                                             params=params, **kwargs)
+            with open_session(**self._session_kwargs) as sess:
```

Comment:
we can't just use `requests.request` because the session [needs the httpadapter for retries](https://github.com/civisanalytics/civis-python/blob/0d9c886570a78bbbdb2f45a53c3ca660db6f89ce/civis/_utils.py#L58-L61), right?

Reply:
I was wondering about this myself -- it looks like we're losing the performance benefits of sessions if we're opening and closing them with each requests, so I was wondering if we could stop using them entirely to avoid a little of the complexity they bring.

Reply:
We could also go the other direction and make the session a singleton so all requests would share the same connection

Reply:
@mheilman , yes, but also, the `request` function has exactly two lines:
```python
    with sessions.Session() as session:
        return session.request(method=method, url=url, **kwargs)
```

It doesn't look like creating a new `Session` has much of a performance penalty. I don't think the Civis API supports querying at a rate high enough where it would matter.

Reply:
> It doesn't look like creating a new `Session` has much of a performance penalty. I don't think the Civis API supports querying at a rate high enough where it would matter.

I meant to refer to the performance benefits of re-using session objects, not the cost of creating new session objects.